### PR TITLE
Refactor func test utils

### PR DESCRIPTION
**Pull request by olpipi**
### Details:
 - Eliminate code duplication in func and common test utils
 - Remove unused code

### Tickets:
 - [CVS-116636](https://jira.devtools.intel.com/browse/CVS-116636)

### CODE REVIEW

**Review by p-durandin — APPROVED** [review]

File: src/plugins/intel_cpu/thirdparty/mlas
Comment (ilyachur) [inline src/plugins/intel_cpu/thirdparty/mlas:1]: Why did you update submodule? Please revert this change

Comment (olpipi) [reply src/plugins/intel_cpu/thirdparty/mlas:1]: This is probably a rebase issue. I will remove it

Comment (olpipi) [reply src/plugins/intel_cpu/thirdparty/mlas:1]: done
